### DISCUSSION TITLE
Error using threejs: Error: No intl context found. Have you configured the provider?

### DISCUSSION THREAD

**Discussion by elblogbruno** [amannn/next-intl#773] [Q&A]
Hello everyone!
I am trying to implement next-intl on my project with nextjs14. I use threejs for creating a 3d interface when I am on desktop and normal ui when I am on mobile phone.
When I am on desktop I have this issue with my header component:
![image](https://github.com/amannn/next-intl/assets/10481058/e56a580f-93de-47be-b727-ea8b9c9d81cf)

`Error: No intl context found. Have you configured the provider?`

However on phone view, it works perfectly:
![image](https://github.com/amannn/next-intl/assets/10481058/056ba601-1296-4b21-b984-e02f4f4d3fd7)

what could be the issue?

Here is my header component. It is used on normal 2d view aswell as threejs, on  threejs it is only wrapped with threejs components to make it be on 3d space:

```
// "use client"
import React from 'react';


import {NextIntlClientProvider, useLocale, useMessages} from 'next-intl';
import ClientHeader from '../../components/portfolio/client_header';


export default function Header({ isDesktop, setBg, onButtonClick }: { isDesktop: boolean; setBg:boolean; onButtonClick: (message: string) => void }) {
    const messages = useMessages();
    const locale = useLocale();
        
    return (
        <NextIntlClientProvider  messages={messages} locale={locale}>
            <ClientHeader isDesktop={isDesktop} onButtonClick={onButtonClick} setBg={setBg}  />
        </NextIntlClientProvider>
    );
};

// export default Header;
```
thanks

**Comment by elblogbruno** [comment]
![image](https://github.com/amannn/next-intl/assets/10481058/dc9efb25-05e9-496f-a191-33f569a6346c)
I fixed it by wrapping again the header component with NextIntlClientProvider:

![image](https://github.com/amannn/next-intl/assets/10481058/775f8af8-5952-4cd9-b96e-70bf6cdd8f81)

**Reply by amannn** [reply]
Sounds right to me! My guess would be that you're rendering `Header` twice, once from a Server Component and once from a Client Component?

The site looks super nice in the screenshot btw., let me know when it's online!

**Reply by elblogbruno** [reply]
Thank you! I'll let you know of course! By the moment I am trying to make all the puzzle work together, but it is near the end!

**Reply by elblogbruno** [reply]
![image](https://github.com/amannn/next-intl/assets/10481058/8232f768-96f5-4d71-aa3b-0221acdb5dbd)
I have reused the locale switcher component from examples, but I am getting this issue when using threejs fiber. Might it be related to next-intl or its a problem with threejs?

**Reply by amannn** [reply]
Likely the Three.js renderer, yes. Maybe you can move the `useRouter` call to a component that is rendered by Next.js, potentially passing down a callback.
